src/main/java/mate/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package mate.service;
+
+import java.util.Optional;
+import mate.exception.AuthenticationException;
+import mate.lib.Inject;
+import mate.lib.Service;
+import mate.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverService.findByLogin(login);
+        if (driver.isEmpty()) {
+            throw new AuthenticationException("Login or Password was incorrect");
+        }
+        if (driver.get().getPassword().equals(password)) {
+            return driver.get();
+        }
```

Comment:
Both in AuthenticationServiceImpl and in AuthenticationFilter, use only one if block if possible.